src/dashboard/dashboard.controller.ts

```diff
@@ -0,0 +1,37 @@
+import { DashboardService } from '@/dashboard/dashboard.service';
+import { GetUserAmountDto } from '@/dashboard/dto/getAllUserAmount.dto';
+import { Role } from '@/decorator/roles.decorator';
+import { ParseMongoDatePipe } from '@/pipes/mongo-date.pipe';
+import { UserRole } from '@/schema/users.schema';
+import { Controller, Get, HttpCode, HttpStatus, Param } from '@nestjs/common';
+import {
+  ApiBearerAuth,
+  ApiOperation,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+
+@Controller({ path: 'dashboard', version: '1' })
+@ApiTags('dashboard')
+@ApiBearerAuth()
+export class DashboardController {
+  constructor(private readonly dashboardService: DashboardService) {}
+
+  @ApiResponse({
+    status: HttpStatus.OK,
+    type: () => GetUserAmountDto,
+    description: 'Total registered users',
+  })
+  @ApiOperation({
+    summary: 'Get total registered users',
+  })
+  @Get('/customer_report/:date')
+  @HttpCode(HttpStatus.OK)
+  @Role(UserRole.Owner)
+  async getDashboard(@Param('date') date: number) {
+    const parseDate = new ParseMongoDatePipe();
+    return await this.dashboardService.getAllUserAmount(
+      parseDate.transform(date),
+    );
```

Comment:
Your can use pipetransform (parseDate) inside @param see https://docs.nestjs.com/pipes#binding-pipes

Reply:
i try it already but in path its number of unix timestamp but in service required iso date time
if i put `date: Date` in `@Param` like `@Param('date', new ParseMongoDatePipe()) date: Date` it will not cause any error but it cant put number in path
![image](https://github.com/meeorder/meeorder-backend/assets/86043260/4ba51726-55ac-4637-bcf4-da1c4d801afe)

Reply:
but i fix it abit

Reply:
OK Typescript is stupid LGTM!